extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesClientApi.java

```diff
@@ -19,15 +19,30 @@
 
 package org.apache.druid.k8s.overlord.common;
 
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
 import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
 
 // Wraps all kubernetes api calls, to ensure you open and close connections properly
 public interface KubernetesClientApi
 {
   <T> T executeRequest(KubernetesExecutor<T> executor) throws KubernetesResourceNotFoundException;
 
+  <T> T executePodCacheRequest(KubernetesInformerExecutor<T, Pod> executor);
+
+  <T> T executeJobCacheRequest(KubernetesInformerExecutor<T, Job> executor);
+
   // use only when handling streams of data, example if you want to pass around an input stream from a pod,
   // then you would call this instead of executeRequest as you would want to keep the connection open until you
   // are done with the stream.  Callers responsibility to clean up when using this method
   KubernetesClient getClient();
+
+  SharedIndexInformer<Pod> getPodInformer();
+
+  SharedIndexInformer<Job> getJobInformer();
+
+  long getInformerResyncPeriodMillis();
+
+  KubernetesResourceEventNotifier getEventNotifier();
```

Comment:
Instead of returning the event notifier itself, we should expose the wait methods, `waitForJobChange()`, `waitForPodChange()`.